beacon-chain/core/state/transition.go

```diff
@@ -29,6 +29,10 @@ import (
 // processFunc is a function that processes a block with a given state. State is mutated.
 type processFunc func(context.Context, iface.BeaconState, *ethpb.SignedBeaconBlock) (iface.BeaconState, error)
 
+var processDepositsFunc = func(ctx context.Context, s iface.BeaconState, blk *ethpb.SignedBeaconBlock) (iface.BeaconState, error) {
+	return b.ProcessDeposits(ctx, s, blk.Block.Body.Deposits)
+}
```

Comment:
Even though you got rid of one nasty construction, now you have to define this, which is even more artificial in my opinion. But I guess if it helps Altair, then it's all right.

Reply:
Yes. This is expected. The pros far outweights the cons. We should honestly get rid of this ProcessOperation to be `func(ctx context.Context, s iface.BeaconState, blk *ethpb.SignedBeaconBlock) (iface.BeaconState, error)`. This is not going to scale in the long run